### ISSUE TITLE
multi-gpu training

### ISSUE THREAD

**Issue by LiliMeng** [NVIDIA/flownet2-pytorch#44]
Hi,

     Thanks a lot for providing the open-source code!
     It seems that the multi-gpu (I set the argument num_gpus to be 4 , so 4 GPUs are used) is not faster than the single GPU. 
    Could you provide some suggestions?
    Highly appreciate your time and help!

Best,
Lili

**Comment by fitsumreda** [comment]
@LiliMeng could you give more details. what batch size was used, and which architecture did you try?

**Comment by qiuzhongwei-USTB** [comment]
@fitsumreda  I trained flownet2 model with 1 GPU. there is one mistake like the follow picture. but it runs in epoch 0. it go wrong when validated after trained in 1 epoch.
![image](https://user-images.githubusercontent.com/34059431/42873573-e8713d10-8ab1-11e8-86f8-45457978601a.png)

in other hands, when I run this command to trained with 4 GPU, it can't work.
python main.py --start_epoch 0 --total_epochs 10 --batch_size 8 --model FlowNet2 --loss=L1Loss --optimizer=Adam --optimizer_lr=1e-10 --number_workers 0 --number_gpus 4 \
--training_dataset MpiSintelFinal --training_dataset_root /var/storage/shared/pnrsy/sys/jobs/application_1531419344053_9844/flownet2_data/training  \
--validation_dataset MpiSintelClean --validation_dataset_root /var/storage/shared/pnrsy/sys/jobs/application_1531419344053_9844/flownet2_data/training --resume /var/storage/shared/pnrsy/sys/jobs/application_1531419344053_9844/flownet2_data/pre-trained_model/FlowNet2_checkpoint.pth


could you help me to solve this problem?
